tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -26,17 +28,26 @@
 
 namespace nvfuser {
 
+int64_t D = 1, B = 2, E = 768, H = 12, S = 128;
```

Comment:
While I understand you made them global so DistributedTransformerTest can modify their values, these global variables have external linkage and can conflict with variables of the same name in other files. AFAICS, they are only used in DistributedTransformerTest, so I'd make them class members.

Reply:
Is moving the globals into an anonymous namespace sufficient to avoid the global linkage? I avoided putting them as class members so that the helper functions don't have to use `DistributedTransformerTest::D`. This is purely cosmetic though so it's not a must have

Reply:
> Is moving the globals into an anonymous namespace sufficient to avoid the global linkage? 

Yes. However, even if they are in an anonymous namespace, they are still globals that can potentially be modified anywhere in the file, making code harder to reason about. 

After reading the PR again, do we really need to modify E and H? We could gtest_skip when they are not divisible by D. We could also fix H-per-device (so it's a constant) and compute H as `num_devices * H-per-device` etc.

Reply:
Yes, I think it's fine to just skip the tests when E isn't divisible by D. In most cases this is fine (even numbers up to 8) that way everything but D is a constant